component/src/test/java/io/siddhi/extension/io/nats/sink/NATSSinkTestCase.java

```diff
@@ -323,5 +339,57 @@ public void testMissingMappingAnnotation() {
             }
         }
     }
+
+    /**
+     * Test for configure the NATS Sink to publish the message to a NATS-streaming subject.
+     */
+    @Test
+    public void testNatsProtobuf() throws InterruptedException, NoSuchMethodException,
+            InvocationTargetException, IllegalAccessException {
+        ResultContainer resultContainer = new ResultContainer(2, 3);
+        NATSClient natsClient = new NATSClient("test-cluster", "stan_test1", "nats://localhost:"
+                + port, resultContainer);
+        natsClient.connect();
+        SiddhiManager siddhiManager = new SiddhiManager();
+
+        String inStreamDefinition = "@App:name('Test-plan1')\n"
+                + "@sink(type='nats', " +
+                "@map(type='protobuf', class='io.siddhi.extension.io.nats.utils.protobuf.Person'), "
+                + "destination='nats-sink-test10', "
+                + "client.id='test-plan10-siddhi',"
+                + "bootstrap.servers='" + "nats://localhost:" + port + "', "
+                + "cluster.id='test-cluster'"
+                + ")"
+                + "define stream inputStream (nic long, name string);";
+
+        SiddhiAppRuntime executionPlanRuntime = siddhiManager.createSiddhiAppRuntime(inStreamDefinition);
+        InputHandler inputStream = executionPlanRuntime.getInputHandler("inputStream");
+        executionPlanRuntime.addCallback("inputStream", new StreamCallback() {
+            @Override
+            public void receive(Event[] events) {
+                EventPrinter.print(events);
+                for (int i = 0; i < events.length; i++) {
+                    eventCounter.incrementAndGet();
+                }
+            }
+        });
+        executionPlanRuntime.start();
+        Thread.sleep(100);
+
+        long nic1 = 1222;
+        Person person1 = Person.newBuilder().setNic(nic1).setName("Jimmy").build();
+        byte[] messageObjectByteArray1 = (byte[]) AbstractMessageLite.class
+                .getDeclaredMethod("toByteArray").invoke(person1);
+        long nic2 = 1222;
+        Person person2 = Person.newBuilder().setNic(nic2).setName("Natalie").build();
+        byte[] messageObjectByteArray2 = (byte[]) AbstractMessageLite.class
+                .getDeclaredMethod("toByteArray").invoke(person2);
+        inputStream.send(new Object[] {messageObjectByteArray1});
+        inputStream.send(new Object[] {messageObjectByteArray2});
+
+        Thread.sleep(100);
+        AssertJUnit.assertEquals(eventCounter.get(), 2);
```

Comment:
The sink test logic seems to be incorrect. You are asserting the event count at the stream you putting the events to from the input handler.
You need to assert the event count received at the NATS topic using the NATS client you've created. Use resultContainer.